### PR TITLE
More test

### PR DESCRIPTION
**Pull request by frack113**
Speed  many time to check the W3C field name for webserser
- https://learn.microsoft.com/en-us/windows/win32/http/w3c-logging
- https://www.loganalyzer.net/log-analyzer/w3c-extended.html

### CODE REVIEW

File: rules/web/web_cve_2022_27925_exploit.yml [hunk 1]
```diff
@@ -29,7 +30,7 @@ detection:
     selection_shell:
         cs-uri|contains: '/zimbraAdmin/'
         cs-uri|endswith: '.jsp'
```
Comment (nasbench) [inline rules/web/web_cve_2022_27925_exploit.yml:32]: In the docs you linked  `cs-uri` doesn't exist

File: tools/config/qradar.yml [hunk 2]
```diff
@@ -37,6 +37,7 @@ fieldmappings:
  c-uri: URL
  c-uri-extension: URL
  c-useragent: user_agent
```
Comment (nasbench) [inline tools/config/qradar.yml:39]: Why not remove the old one? It's confusing now.

Comment (frack113) [reply tools/config/qradar.yml:39]: For the proxy category

File: rules/web/web_cve_2022_27925_exploit.yml (hunk 1 above)
Comment (frack113) [reply rules/web/web_cve_2022_27925_exploit.yml:32]: `cs-uri-query` 👍

File: tests/test_logsource.py [hunk 3]
```diff
@@ -143,20 +143,26 @@ def check_service(product,service,fieldname) -> bool:
...
                                 print(
                                     Fore.RED + "Rule {} has the invalid field <{}> for service <{}>".format(file, field,full_logsource['service']))
                                 files_with_fieldname_issues.append(file)                        
-
+                else:
+                    if full_logsource['category']:
+                        for field in self.get_detection_field(detection):
+                            if not check_category("empty",full_logsource['category'],field):
+                                print(
+                                    Fore.RED + "Rule {} has the invalid field <{}> forcategory <{}>".format(file, field,full_logsource['category']))
```
Comment (nasbench) [inline tests/test_logsource.py:164]:
```suggestion
                                    Fore.RED + "Rule {} has the invalid field <{}> for category <{}>".format(file, field,full_logsource['category']))
```

**Review by nasbench — APPROVED** [review]

**Pull request merged**